src/solvers/dgsem_tree/indicators.jl

```diff
@@ -215,6 +215,93 @@ const IndicatorLoehner = IndicatorLöhner
     return num / den
 end
 
+"""
+    IndicatorIDP(equations::AbstractEquations, basis;
+                 positivity=false,
+                 variables_cons=[],
+                 positivity_correction_factor=0.1)
+
+Subcell invariant domain preserving (IDP) limiting used with [`VolumeIntegralSubcellLimiting`](@ref)
+including:
+- positivity limiting for conservative variables (`positivity`)
+
+The bounds are calculated using the low-order FV solution. The positivity limiter uses
+`positivity_correction_factor` such that `u^new >= positivity_correction_factor * u^FV`.
+
+## References
+
+- Rueda-Ramírez, Pazner, Gassner (2022)
+  Subcell Limiting Strategies for Discontinuous Galerkin Spectral Element Methods
+  [DOI: 10.1016/j.compfluid.2022.105627](https://doi.org/10.1016/j.compfluid.2022.105627)
+- Pazner (2020)
+  Sparse invariant domain preserving discontinuous Galerkin methods with subcell convex limiting
+  [DOI: 10.1016/j.cma.2021.113876](https://doi.org/10.1016/j.cma.2021.113876)
+
+!!! warning "Experimental implementation"
+    This is an experimental feature and may change in future releases.
+"""
+struct IndicatorIDP{RealT <: Real, Cache} <: AbstractIndicator
+    positivity::Bool
+    variables_cons::Vector{Int}                     # Impose positivity for conservative variables
```

Comment:
With the last commit I changed the selection of conservative variables to an array of their specific numbers. Example: `variables_cons=[1]` instead of `variables_cons=(Trixi.density,)`).
This is required for following PRs.
The upcoming nonlinear variables are still selected with a Tuple of functions. Is it possible to use a tuple here as well? It must have an undefined length. So, rn we would have different data types for the variables.

Alternatively, I could change the call of non-linear variables to an array as well.